org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaModelManager.java

```diff
@@ -1258,7 +1257,7 @@ public static IJavaElement determineIfOnClasspath(IResource resource, IJavaProje
 	/**
 	 * Infos cache.
 	 */
-	private JavaModelCache cache;
+	private final JavaModelCache cache = new JavaModelCache();
```

Comment:
Interesting, this is actually "back to the roots". Original code had this until commit af33da7216734bcb2ae0e5dbadaa836c42812db8 that moved init after init of debug options.

Reply:
The JavaModelCache constructor already uses the debug options, therefore the debug options have been initialized before.